user/client_test.go

```diff
@@ -44,8 +66,50 @@ func TestUserGetMetadata(t *testing.T) {
 	assert.Equal(t, "0x4B73C58370AEfcEf86A6021afCDe5673511376B2", meta.PublicAddress)
 }
 
+func TestUserGetMetadataWithWallet(t *testing.T) {
+	srv := createServerSuccess(t, &testSuccessDataWithWallets)
+	defer srv.Close()
+
+	// Replace host url to test one.
+	client := magic.NewDefaultClient()
+	client.SetHostURL(srv.URL)
+
+	uClient := NewUserClient(testSecret, client)
+
+	meta, err := uClient.GetMetadataByTokenAndWallet(testDIDToken, wallet.SOLANA)
+	require.NoError(t, err, "can't create new token")
+
+	assert.Equal(t, "user@email.com", meta.Email)
+	assert.Equal(t, "did:ethr:0x4B73C58370AEfcEf86A6021afCDe5673511376B2", meta.Issuer)
+	assert.Equal(t, "0x4B73C58370AEfcEf86A6021afCDe5673511376B2", meta.PublicAddress)
+	assert.NotNil(t, meta.Wallets)
+	assert.Equal(t, 1, len(*meta.Wallets))
```

Comment:
There's an `assert.Len` for this too :D `assert.Len(t, mySlice, 3)`

Reply:
Very nice 🙌🏻 Updated!